fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordContext.java

```diff
@@ -205,7 +201,7 @@ protected FDBRecordContext(@Nonnull FDBDatabase fdb,
                 tr.options().setPrioritySystemImmediate();
                 break;
             default:
-                throw new RecordCoreArgumentException("unknown priority level " + priority);
+                throw new RecordCoreArgumentException("unknown priority level " + config.getPriority());
         }
 
         // Set the transaction timeout based on the config (if set) and the database factory otherwise
```

Comment:
The `timeoutMillis` field remains _only_ for the `STABLE` `getTimeoutMillis` API in the case where the config has default and the database factory value changes _after_ the context is created.

If we were willing to have `getTimeoutMillis` do what `getTimeoutMillisToSet` does now and accept that edge case, it could be removed as well.

Reply:
I think leaving it like this (at least for now) seems fine to me. It also seems clear (from the comments and the code) why it's separated out from the config, so it seems maintainable